enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,250 @@
+communication ingress flows matrix.
+
+- A communication matrix describing the expected flows of incoming traffic will 
+  be included in every OpenShift release documentation.
+
+- A new `oc` command will be added to generate a current snapshot of known 
+  listening ports in a running cluster, `oc adm communication-matrix generate`.
+
+- A new option will be added to OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### Workflow Description
+
+An OpenShift administrator would like to get an accurate and up-to-date OpenShift 
+communication ingress flows matrix.
+
+- The admin reviews OpenShift release documentation to get the included communication
+  matrix describing the expected flows of incoming traffic.
+
+- The admin uses the OpenShift command-line interface (CLI) to generate an up-to-date 
+  communication matrix using the following command `oc adm communication-matrix generate`.
+
+- Optionally, the admin uses the OpenShift web console to generate an up-to-date
+  communication matrix.
+
+### API Extensions
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Out of scope for this proposal.
+
+#### Standalone Clusters
+Out of scope for this proposal.
+
+#### Single-node Deployments or MicroShift
+
+The communication matrix can be generated on single-node deployments.
+
+MicroShift is out of scope for this proposal.
+
+### Implementation Details/Notes/Constraints
+
+A new helper module will be added to `openshift/library-go` to create a communication
+ingress flows matrix from the existing `EndpointSlices` in the OpenShift cluster.
+
+`EndpointSlices` group network endpoints together by unique combinations of 
+protocol, port number, and `Service` name and therefore can be used as the 
+source of truth for cluster traffic. For that to work, every application that 
+attracts ingress traffic running in the cluster should have a `Service` object,
+the control plane automatically creates `EndpointSlices` for any `Service` that 
+has a selector specified. `EndpointSlices` can also be directly created when needed
+or managed by other entities or controllers.
+
+To generate the communication matrix the `EndpointSlices` in the cluster should 
+be filtered by `Service` type `NodePort` and `LoadBalancer` as there are also 
+`EndpointSlices` in the cluster for internal `Services`. The node hosted services 
+such as sshd, rpc, etc. are currently missing `EndpointSlices` (see [Open Questions](#open-questions)).
+
+More info about the Kubernetes `EndpointSlices` resource and management can be found
+in https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#management
+
+The module allows to generate the communication matrix in various formats:
+- YAML/JSON
+- CSV
+- nftables
+
+Each record describes a flow with the following information:
+```
+direction      Data flow direction (currently ingress only)
+protocol       IP protocol (TCP/UDP/etc)
+port           Flow port number
```

Comment:
this is the service port right not the endpoint , also do u plan to suport SCTP` protocol ?
also for ingress firewall setting the rules might also require knowing the srcIP

Reply:
That's the endpoint port
@liornoy can/do we support SCTP protocol? also you have a test to apply firewall rules right? does it require the srcIP?

Reply:
so actually 3 questions here:
1. service port vs endpoint port - it's the endpoint / the one seen on the host network side
2. SCTP - theoretically can be included too, especially in the proposal. impl will focus on TCP/UDP in the first iteration. can add later
3. srcIP - there might be cases where you want to limit a certain service/flow to be accessible only from specific ip ranges. but that's mostly up to the customer. so short answer is for now it's binary yes/no

Reply:
If the tool is to be used by customers to generate their matrix, they would like it to include their SCTP ports too.

for the one we are generating for OCP, you are right.